### ISSUE TITLE
[🐞] `useContextProvider()`: Does not accept strings as input types.

### ISSUE THREAD

**Issue by TheAyes** [BuilderIO/qwik#5793]
### Which component is affected?

Qwik Runtime

### Describe the bug

I'm trying to pass a scoped context to a few children to be able to determine the children size based on the context.

### Reproduction

(currently impossible due to blockages by my company. If required I can get one whenever I get home)

### Steps to reproduce

My Structure looks a bit like this:
```tsx
type ButtonSize = "small" | "normal" | "large";

const buttonSizeContext = createContextId<ButtonSize>("MyApp.ButtonSizeContext");

const Button = component$(() => {
  return ( <button/> );
});

const ButtonGroup = component$(() => {
  const size = useContext(buttonSizeContext)

  return (
    <div class={}>
      <Slot>
    </div>
  );
});

const ButtonSizeProvider = component$<{size: ButtonSize}>(({size: ButtonSize}) => {
  
  // TS2344: Type string does not satisfy the constraint object
  // Type string is not assignable to type object
  //                 vvvvvvvvvv
  useContextProvider<ButtonSize>(buttonSizeContext, "small")

  return (
    <div>
      <Slot/>
    </div>
  );
});

const Page = component$(()) => {
  return (
    <ButtonGroup>
      <ButtonSizeProvider size="large">
        <Button/>
        <Button/>
        <Button/>
        <Button/>
        <Button/>
      </ButtonSizeProvider>
      <ButtonSizeProvider size="normal">
        <Button/>
        <Button/>
        <Button/>
        <Button/>
        <Button/>
      </ButtonSizeProvider>
    </ButtonGroup>

    <ButtonGroup>
      <ButtonSizeProvider size="large">
        <Button/>
        <Button/>
        <Button/>
        <Button/>
        <Button/>
      </ButtonSizeProvider>
      <ButtonSizeProvider size="normal">
        <Button/>
        <Button/>
      </ButtonSizeProvider>
      <ButtonSizeProvider size="small">
        <Button/>
        <Button/>
        <Button/>
      </ButtonSizeProvider>
    </ButtonGroup>

    <ButtonGroup>
      <ButtonSizeProvider size="normal">
        <Button/>
        <Button/>
        <Button/>
      </ButtonSizeProvider>
    </ButtonGroup>
  )
});
```

As you can see I have an error in the type generic where the useContextProvider Function expects the type generic to extend the object type. In your Documentation however you do the very same thing except for not defining a type at all:
```tsx
import { component$, useStore, useContextProvider } from '@builder.io/qwik';
 
export const Parent = component$(() => {
 
  const qwikCityObject = useStore<GenericType>({
    ...
  });
 
  useContextProvider(QwikCityContext, qwikCityObject);
  useContextProvider(PlainArrayContext, [1, 2, 3])
  useContextProvider(AppNameContext, "My Qwik App")
 
  return (
    <Children />
  );
});
```

I've also tried removing the type generic but this makes the context parametter error:
```
TS2345: Argument of type ContextId<unknown> is not assignable to parameter of type ContextId<object>
Type unknown is not assignable to type object
```

### System Info

```shell
System:
    OS: Windows 11 10.0.22631
    CPU: (32) x64 13th Gen Intel(R) Core(TM) i9-13900
    Memory: 17.23 GB / 31.65 GB
  Binaries:
    Node: 21.6.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.2.5 - C:\Program Files\nodejs\npm.CMD
    pnpm: 8.14.3 - ~\AppData\Local\pnpm\pnpm.CMD
  Browsers:
    Edge: Chromium (121.0.2277.83)
    Internet Explorer: 11.0.22621.1
  npmPackages:
    @builder.io/qwik: ~1.3.1 => 1.3.5
    @builder.io/qwik-city: ~1.3.1 => 1.3.5
    undici: ^5.22.0 => 5.28.2
    vite: ^5.0.0 => 5.0.12
```

### Additional Information

Most certainly this error originates from your function type declaration here:
```ts
export declare const useContextProvider: <STATE extends object>(context: ContextId<STATE>, newValue: STATE) => void;
```

For now I can work around this issue by wrapping my string in an object. This is however undesireable and I would like to see a fix sometime in the future.

**Comment by wmertens** [comment]
Can you use `<any>` on the provider? That should get rid of the type error and it might just work after that or there might be serialization issues.

**Comment by TheAyes** [comment]
> Can you use `<any>` on the provider? That should get rid of the type error and it might just work after that or there might be serialization issues.

For now I've just wrapped it in an object to get rid of the type error:
```tsx
useContextProvider<{ size: ActionButtonSize }>(
  actionButtonSizeContext,
  { size }
);
```
which I then immediatly access in my button component like this:
```tsx
const size = useContext(actionButtonSizeContext).size;
```

For now it works but it'd be nice to see this resolved in the future :)

**Comment by wmertens** [comment]
Fixed in #5914
